setup.sh

```diff
@@ -1,31 +1,93 @@
 #! /bin/sh
```

Comment:
I suggested this in the past but for reasons I don't remember it was never changed. This allows me to run the script without directly mentioning `bash` (it uses features not available in standard `sh`)

```suggestion
#! /bin/bash
```